.github/workflows/build.yml

```diff
@@ -0,0 +1,38 @@
+name: Test and Build React
+
+on:
+    push:
+
+    pull_request:
+        branches: ['master']
+
+    workflow_dispatch:
+
+jobs:
+    build:
+        runs-on: ubuntu-latest
+
+        steps:
+            - uses: actions/checkout@v3
+
+            - uses: actions/setup-node@v3
+              with:
+                  cache: 'npm'
+
+            - run: npm ci
+
+            - name: ESLint check
+              run: npm run lint:check
+
+            - name: Prettier check
+              run: npm run format:check
+
+            - name: Build app
+              run: npm run build
```

Comment:
This is a personal preference thing - but my opinion would be to try building the app first, and then running linting and formatting checks.

 The reason behind this is that failures to build are going to be more critical than linting/formatting errors - if a build is failing and the user hasn't noticed, I would find this much more significant than a lint/formatting failure and would want to see that failure before any of the others. Since the first failing task will stop the build, I think we should check things in order of importance, which I think would be `build` > `lint` > `format`.

What are your thoughts on this?

Reply:
That makes sense, I didn't think about it that way. Although, I think `lint` is actually part of the build process, so maybe we just do `build` > `format`?

Reply:
oh, interesting. Does `react-scripts build` also do linting?

To be safe, I'd still recommend having a stage for each (in case it doesn't follow the same ESLint linting rules), and to make it clear where failures are occurring

Reply:
Problem is if ESLint fails the build it will always happen in the build step and not the eslint step, so the step doesn't do much

Reply:
But I guess it doesn't hurt anything, so might as well put it in